crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -1299,28 +1431,49 @@ impl Segment2d {
         rotated.translated(point)
     }
 
-    /// Compute a new segment, based on the original segment rotated around its center
+    /// Compute the segment rotated around its own center.
     #[inline(always)]
     pub fn rotated_around_center(&self, rotation: Rot2) -> Segment2d {
         self.rotated_around(rotation, self.center())
     }
 
-    /// Get the segment with its center at the origin
+    /// Compute the segment with its center at the origin, keeping the same direction and length.
     #[inline(always)]
     pub fn centered(&self) -> Segment2d {
         let center = self.center();
         self.translated(-center)
     }
 
-    /// Get the segment with a new length
+    /// Compute the segment with a new length, keeping the same direction and center.
     #[inline(always)]
     pub fn resized(&self, length: f32) -> Segment2d {
         let offset_from_origin = self.center();
-        let centered = self.centered();
+        let centered = self.translated(-offset_from_origin);
         let ratio = length / self.length();
         let segment = Segment2d::new(centered.point1() * ratio, centered.point2() * ratio);
         segment.translated(offset_from_origin)
     }
+
+    /// Swaps the two endpoints of the line segment.
+    #[inline(always)]
+    pub fn swap(&mut self) {
+        let [a, b] = &mut self.vertices;
+        core::mem::swap(a, b);
+    }
```

Comment:
I found the `&mut self` and the name a bit surprising. Could consider this?

```suggestion
    pub fn reversed(&self) -> Self {
        Segment2d::new(self.point2(), self.point1())
    }
```

Reply:
You could also consider `flip(&mut self)`.